goldstone/core/models.py

```diff
@@ -882,31 +801,34 @@ def clouddata():
         # Availability Zone for each one.
         result = []
 
+        # For every found host...
         for host in hosts:
             parsed_name = Host._parse_host_name(host["host_name"])[0]
 
             if all(x["host_name"] != parsed_name for x in result):
-                # This is a new entry for the result set.
+                # This is a new entry for the result set. Set the host_name
+                # value, and concoct a unique id for it.
                 host["host_name"] = parsed_name
+                host[cls.unique_cloud_id_key] = cls.unique_cloud_id(host)
+
+                # We don't want or need this key.
                 del host["service"]
```

Comment:
This implies a relation between the host and a 'service' that runs on it.  IIRC, the openstack model is a bit flaky here, in that it will give you multiple host entries with the same name, but a different service.  

```
(goldstone-server)[~/devel/goldstone-server]$ nova service-list                                                                                [issue_22]
+----+------------------+------------------+----------+---------+-------+----------------------------+-----------------+
| Id | Binary           | Host             | Zone     | Status  | State | Updated_at                 | Disabled Reason |
+----+------------------+------------------+----------+---------+-------+----------------------------+-----------------+
| 1  | nova-consoleauth | base.pepple.info | internal | enabled | up    | 2015-06-22T14:53:52.000000 | -               |
| 2  | nova-scheduler   | base.pepple.info | internal | enabled | up    | 2015-06-22T14:53:53.000000 | -               |
| 3  | nova-conductor   | base.pepple.info | internal | enabled | up    | 2015-06-22T14:53:52.000000 | -               |
| 4  | nova-compute     | base.pepple.info | nova     | enabled | up    | 2015-06-22T14:53:58.000000 | -               |
| 5  | nova-cert        | base.pepple.info | internal | enabled | up    | 2015-06-22T14:53:52.000000 | -               |
+----+------------------+------------------+----------+---------+-------+----------------------------+-----------------+
```

This is an overload of the definition of service (the other is a Keystone service) that I'm not sure we have represented well in our model.  Let me if `host --> nova service` is missing from the model and I'll ticket it separately.